filebeat/harvester/log.go

```diff
@@ -112,9 +123,16 @@ func (h *Harvester) createEvent() *input.FileEvent {
 }
 
 // sendEvent sends event to the spooler channel
-func (h *Harvester) sendEvent(event *input.FileEvent) {
-	// FIXME: This will send on closed channel if not properly exited
-	h.SpoolerChan <- event // ship the new event downstream
+// Return false if event was not sent
+func (h *Harvester) sendEvent(event *input.FileEvent) bool {
+	select {
+	case <-h.done:
+		return false
+	default:
+		h.SpoolerChan <- event // ship the new event downstream
```

Comment:
I would expect so see another case statement instead of a default block for this channel operation. Any reason to not write:

```
case h.SpoolerChan <- event:
```